Migrate to arcade-ai v2 guide (#326)
ArcadeAI/docs: pages/home/migrate-to-v2.mdx

# Migration Guide

## Important Notice

> If your project does not have its dependency on `arcade-ai` pinned to the v1 major version, you must pin it to the v1 major version immediately to prevent potential breakages.

Reviewer: [nit] there's probably some graphical treatment we could do here - alert?

## Release Timeline

> Version 2.0.0 will be released on **Tuesday, June 17th during PST morning hours**. Make sure to pin your project's dependency on `arcade-ai` to the v1 major version before this date to avoid any disruptions.

Reviewer: ```suggestion
Version 2.0.0 of the Arcade CLI was released on **Tuesday, June 17th**. Make sure to pin your project's dependency on `arcade-ai` to the v1 major version before this date to avoid any disruptions.
```

## Package Changes
With the release of version 2.0.0, the [arcade-ai PyPI package](https://pypi.org/project/arcade-ai/) has been divided into three distinct packages. Previously, `arcade-ai` was a monolithic package that included:
- Tool Development Kit (TDK)
- Core execution engine and catalog functionality
- FastAPI/MCP server/worker components
- CLI functionality
- Evaluation framework

> Now, the `arcade-ai` package serves as Arcade.dev's CLI, allowing you to create 'starter' toolkits, serve and deploy toolkits, chat, and more via the command line. Additionally, two new packages have been introduced: `arcade-serve` and `arcade-tdk`.

Reviewer: Add more of the features! 
* Less dependencies
* Smaller size
* etc
